add_me_to_your_PATH/sysimage.jl

```diff
@@ -1,29 +1,13 @@
-using Pkg
+using Pkg, PackageCompiler, UUIDs
 
-const project = Pkg.project()
-
-const modulefile = joinpath(dirname(project.path), "src", "$(project.name).jl")
+function main()
+    project = Pkg.project()
 
-if project.ispackage && !isfile(modulefile)
-    @info "Project is a package, but has no src/\$JULIA_PROJECT.jl, creating an empty one."
-    mkpath(dirname(modulefile))
-    open(modulefile, "w") do io
-        println(io, "module $(project.name)")
-        println(io, "end")
+    # Skip generating a system image when there are no dependencies, or all deps are stdlibs.
+    if isempty(project.dependencies) || values(project.dependencies) ⊆ keys(Pkg.Types.stdlibs())
+        exit(0)
     end
-end
```

Comment:
this seems like it might be potentially breaking right?  I agree that it's probably fine to remove since this is probably too magical; but maybe we should provide some more guardrails here (throw an informative error message or something)

Reply:
The creation of the empty project file has been dead code for some time. I do remember Julia complained about not having this file in the past when doing precompilation but I believe those would be on older versions of Julia we no longer care about.

As for the early exit  this is the behavior this script did before so it's definitely not breaking. I can add an informative warning though.

Reply:
Added a warning